### PR TITLE
Update CLI project config tests with latest templates

### LINKED ISSUE #9500: CLI tests use outdated project files
### Problem Description

We've made several template changes (specifically the new PackageReference work), but some of our CLI e2etests contain project files that were generated a while ago.

We should update the test assets so we can confirm that the CLI code is still being tested appropriately.

### Steps To Reproduce

yarn test

### Expected Results

_No response_

### CLI version

npx react-native --version

### Environment

```markdown
npx react-native info
```

### Target Platform Version

_No response_

### Target Device(s)

_No response_

### Visual Studio Version

_No response_

### Build Configuration

_No response_

### Snack, code example, screenshot, or link to a repository

_No response_

### PR DESCRIPTION
**Pull request by jonthysell**
This PR updates the project config tests to remove outdated
test configurations and hard-coded output files.

I've also consolidated the project generation code to reduce the
amount of duplication. This has made it easier to add tests for our
experimental properties: `UseExperimentalNuget`, `UserWinUI3`, and
`UseHermes`.

Closes #9500

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/9623)

### CODE REVIEW

File: packages/@react-native-windows/cli/src/e2etest/projectConfig.utils.ts [hunk 1]
```diff
@@ -12,37 +12,37 @@ export async function tryMkdir(dir: string): Promise<void> {
...
-    namespace: 'WithWinUI3',
+    name,
+    namespace: name,
     useMustache: true,
     projectGuidUpper: testProjectGuid,
     projectGuidLower: testProjectGuid.toLowerCase(),
-    useWinUI3: false,
-    useHermes: false,
-    useExperimentalNuget: false,
+    useWinUI3: !!useWinUI3,
```
Comment (chiaramooney) [inline packages/@react-native-windows/cli/src/e2etest/projectConfig.utils.ts:28]: Out of curiosity what is the `!!` syntax for?

Comment (NickGerleman) [reply packages/@react-native-windows/cli/src/e2etest/projectConfig.utils.ts:28]: It converts something "truthy" or "falsey" to a boolean. E.g."null" or "false" would both become "false". 

The same trick is common in C++ when interoperating with "C" APIs that use "1" and "0" for true and false.

Comment (NickGerleman) [reply packages/@react-native-windows/cli/src/e2etest/projectConfig.utils.ts:28]: The first negation coerces the value to a boolean (that's the opposite of the value you want), and the second inverts it again.

**Review by chiaramooney — APPROVED** [review]
looks good to me 👍

**Pull request merged**